### PR TITLE
E2e test upgrade core services

### PR DESCRIPTION
**Pull request by shifali19**

### CODE REVIEW

File: test/wait/contrail.go [hunk 1]
```diff
@@ -93,6 +93,32 @@ func (c Contrail) ForPostgresActive(name string) error {
 	return err
 }
 
+// ForPodImageChange is used to wait until Image is updated
+func (c Contrail) ForPodImageChange(kubeClient kubernetes.Interface, name string, newImage string, container string) error {
```
Comment (tomaszmj) [inline test/wait/contrail.go:97]: I would rename it to `WaitForPodImageChange`

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 2]
```diff
@@ -317,6 +332,159 @@ func TestHACoreContrailServices(t *testing.T) {
...
+			err := f.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster1", Namespace: namespace}, instance)
+			assert.NoError(t, err)
+
+			zkContainer := utils.GetContainerFromList("zookeeper", instance.Spec.Services.Zookeepers[0].Spec.ServiceConfiguration.Containers)
+			zkContainer.Image = "registry:5000/common-docker-third-party/contrail/zookeeper:" + targetVersionMap["zookeeper"]
+
+			rmqContainer := utils.GetContainerFromList("rabbitmq", instance.Spec.Services.Rabbitmq.Spec.ServiceConfiguration.Containers)
+			rmqContainer.Image = "registry:5000/common-docker-third-party/contrail/rabbitmq:" + targetVersionMap["rabbitmq"]
+
+			//csContainer := utils.GetContainerFromList("cassandra", instance.Spec.Services.Cassandras[0].Spec.ServiceConfiguration.Containers)
```
Comment (tomaszmj) [inline test/e2e/ha/scale_core_contrail_services_test.go:347]: Maybe add a comment why this is commented out (test fails or something like this)

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 3]
```diff
@@ -317,6 +332,159 @@ func TestHACoreContrailServices(t *testing.T) {
...
+				require.NoError(t, err)
+			})
+
+			t.Run("then Rabbitmq StatefulSet should be ready", func(t *testing.T) {
+				assert.NoError(t, w.ForReadyStatefulSet("hatest-rabbitmq-rabbitmq-statefulset", replicas))
+			})
+
+			t.Run("then Cassandra has updated image", func(t *testing.T) {
+				t.Skip()
+				/*err := wait.Contrail{
```
Comment (tomaszmj) [inline test/e2e/ha/scale_core_contrail_services_test.go:412]: If test is skipped, then it doesn't have to be commented out

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 4]
```diff
@@ -317,6 +332,159 @@ func TestHACoreContrailServices(t *testing.T) {
 				}
 			})
 		})
+		t.Run("when manager resource is upgraded", func(t *testing.T) {
```
Comment (tomaszmj) [inline test/e2e/ha/scale_core_contrail_services_test.go:335]: Why is there an upgrade test in file `scale_core_contrail_services_test.go`? Isn't it tested in `upgrade_core_contrail_services_test.go`?

File: test/e2e/ha/upgrade_core_contrail_services_test.go [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+			devicemanagerContainer := utils.GetContainerFromList("devicemanager", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			dnsmasqContainer := utils.GetContainerFromList("dnsmasq", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			schematransformerContainer := utils.GetContainerFromList("schematransformer", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			servicemonitorContainer := utils.GetContainerFromList("servicemonitor", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			analyticsapiContainer := utils.GetContainerFromList("analyticsapi", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			collectorContainer := utils.GetContainerFromList("collector", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			queryengineContainer := utils.GetContainerFromList("queryengine", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			//statusmonitorContainer := utils.GetContainerFromList("statusmonitor", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
```
Comment (tomaszmj) [inline test/e2e/ha/upgrade_core_contrail_services_test.go:355]: Similar as in other places - if this it commented out due to a bug, I would put a comment

File: test/e2e/ha/scale_core_contrail_services_test.go (hunk 4 above)
Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go:335]: "scale_core_contrail_services_test.go" tests for upgrade of all services at once. I had random results with it for Cassandra. "upgrade_core_contrail_services_test.go" tests for upgrade of one service at a time. This way the tests always failed for Cassandra.

File: test/wait/contrail.go (hunk 1 above)
Comment (shifali19) [reply test/wait/contrail.go:97]: Please take a look at how this function is being used/called. That should convince you about the name.

File: test/e2e/ha/upgrade_core_contrail_services_test.go (hunk 5 above)
Comment (shifali19) [reply test/e2e/ha/upgrade_core_contrail_services_test.go:359]: Done

File: test/e2e/ha/scale_core_contrail_services_test.go (hunk 3 above)
Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go:419]: Done

File: test/e2e/ha/scale_core_contrail_services_test.go (hunk 2 above)
Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go:350]: Done

**Comment by shifali19** [comment]
recheck ha

**Comment by shifali19** [comment]
recheck ha

**Pull request merged**